### PR TITLE
Migrate Mobile Center Distribute update token

### PR DESCRIPTION
**Pull request by clpolet**

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistribute.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -151,7 +154,6 @@
 		046BE64B1F8C1ACE00310886 /* ja */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ja; path = Resources/ja.lproj/AppCenterDistribute.strings; sourceTree = "<group>"; };
 		046BE64C1F8C1AE100310886 /* it */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = it; path = Resources/it.lproj/AppCenterDistribute.strings; sourceTree = "<group>"; };
 		046BE64D1F8C1AF200310886 /* pl */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = pl; path = Resources/pl.lproj/AppCenterDistribute.strings; sourceTree = "<group>"; };
-		048210261FD08C42006824F7 /* pt */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = pt; path = Resources/pt.lproj/AppCenterDistribute.strings; sourceTree = "<group>"; };
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute.xcodeproj/project.pbxproj]: Do you know why it changed this?

File: AppCenterDistribute/AppCenterDistribute/Internals/DataMigration/MSKeychainUtil+DistributeMigration.m [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@implementation MSKeychainUtil (DistributeMigration)
+
++ (void)migrateDistributeData {
+
+  // Migrate Mobile Center update token.
+  NSString *mcServiceName =
+      [NSString stringWithFormat:@"%@.%@", @"MobileCenter", [MS_APP_MAIN_BUNDLE bundleIdentifier]];
+  NSString *mcUpdateToken = [MSKeychainUtil stringForKey:kMSUpdateTokenKey withServiceName:mcServiceName];
+  NSString *acUpdateToken = [MSKeychainUtil stringForKey:kMSUpdateTokenKey];
+  if (!acUpdateToken && mcUpdateToken) {
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/Internals/DataMigration/MSKeychainUtil+DistributeMigration.m]: If both tokens exist, probably we want to delete the one for mobile center.

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 3]
```diff
@@ -66,6 +67,9 @@ - (instancetype)init {
                                selector:@selector(applicationWillEnterForeground)
                                    name:UIApplicationWillEnterForegroundNotification
                                  object:nil];
+
+    // Migrate data from previous versions.
+    [MSKeychainUtil migrateDistributeData];
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Should we do this before accessing any of user defaults? I think this should be in line 52.

Comment (clpolet) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: We can do it earlier yes as it's just model.

File: AppCenterDistribute/AppCenterDistribute/Internals/DataMigration/MSKeychainUtil+DistributeMigration.m (hunk 2 above)
Comment (clpolet) [reply AppCenterDistribute/AppCenterDistribute/Internals/DataMigration/MSKeychainUtil+DistributeMigration.m]: Will do

File: AppCenterDistribute/AppCenterDistribute.xcodeproj/project.pbxproj (hunk 1 above)
Comment (clpolet) [reply AppCenterDistribute/AppCenterDistribute.xcodeproj/project.pbxproj]: It jumped couple of lines below.

Comment (clpolet) [reply AppCenterDistribute/AppCenterDistribute.xcodeproj/project.pbxproj]: XCode as its secrets

**Comment by clpolet** [comment]
So, I couldn't use the classic trick to have the category working since the main class needs to be aware of its categories and that not the case here. Plus, it was kind of worthless to use a category for the keychainUtil since all the methods are at class level, it only adds complexity. Though I felt interesting to introduce a brand new object to centralize whatever migration we will have in the future (user defaults, DB ...). That way if one day we want to remove one of the migration because one particular version of the SDK is no more used then we will no where to look :).

File: AppCenter/AppCenter/Internals/Util/MSKeychainUtil.m [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
 
 @implementation MSKeychainUtil
 
-static NSString *AppCenterKeychainServiceName(void) {
+static NSString *AppCenterKeychainServiceName(NSString *suffix) {
   static NSString *serviceName = nil;
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-    serviceName = [NSString stringWithFormat:@"%@.AppCenter", [MS_APP_MAIN_BUNDLE bundleIdentifier]];
+    serviceName = [NSString stringWithFormat:@"%@.%@", suffix, [MS_APP_MAIN_BUNDLE bundleIdentifier]];
```
Comment (clpolet) [inline AppCenter/AppCenter/Internals/Util/MSKeychainUtil.m]: Oops my bad, it's the other way around, suffix should be the last. Changing it now.

Comment (clpolet) [reply AppCenter/AppCenter/Internals/Util/MSKeychainUtil.m]: Done

File: AppCenter/AppCenter/Internals/Util/MSKeychainUtilPrivate.h [hunk 5]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Util/MSKeychainUtilPrivate.h]: Remove empty lines.

File: AppCenterDistribute/AppCenterDistribute/Internals/MSDistributeDataMigration.m [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#import "MSKeychainUtilPrivate.h"
+#import "MSUtility.h"
+
+@implementation MSDistributeDataMigration
+
++(void)migrateKeychain{
+  
+  // Migrate Mobile Center update token.
+  NSString *mcServiceName =
+  [NSString stringWithFormat:@"%@.%@", [MS_APP_MAIN_BUNDLE bundleIdentifier], @"MobileCenter"];
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/Internals/MSDistributeDataMigration.m]: indent.

File: AppCenterDistribute/AppCenterDistribute/Internals/MSDistributeDataMigration.m [hunk 7]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/Internals/MSDistributeDataMigration.m]: Remove empty line.

**Comment by jaelim-ms** [comment]
Please fix formats.

**Pull request merged**